### ISSUE TITLE
Android 8 issue - Connection drop after some time

### ISSUE THREAD

**Issue by MrVipinVijayan** [nishkarsh/android-wificonnect#2]
Thanks for the code.

I am trying your code in Nexus Android 8. It will connect to WiFi as expected but after 1o seconds, it will drop back to the previous wifi.

**Comment by nishkarsh** [comment]
Hey @MrVipinVijayan, does the network to which you are trying to connect has internet connectivity? In case it doesn't, you might need to use WifiConnectionManager#setBindingEnabled(true) to bind the app to the network to make sure that the data is sent through this network.

In case this is taken care of, could you please list the exact steps that you are following. The sample code should help the most.

**Comment by MrVipinVijayan** [comment]
The Wifi to which I am connecting has internet connectivity (Wifi 2) and it
is open. But I have another wifi (Wifi 1) which is also open and has
internet connectivity.
If I connect to the Wifi2 first manually and then use you code to connect
to Wifi1. It will connect to Wifi1, but after 10-20 seconds, it will bounce
back to Wifi2.
Could you please help me on this?

**Comment by MrVipinVijayan** [comment]
Hi,

  Did you have any suggestion on this?

**Comment by nishkarsh** [comment]
@MrVipinVijayan I need a physical device to test this out and that is why the delay. I will try to get hold of an Android device running Android O and try to reproduce this. You could help me out by providing the sample code that you must have written.

Also if you could test the same code on other devices running a different version of Android OS and provide your observations, that should also help a lot.

**Comment by MrVipinVijayan** [comment]
The same code is working on other devices (Such as samsung with Android
7).  When i try on a Nexus Device with Android Oreo, I am getting this
issue. It connects initially, then disconnects and switch to previous
networks probably after 10 seconds.

**Comment by nishkarsh** [comment]
@MrVipinVijayan That helps. Also to know if the issue occurs on a specific device (Nexus) or a specific OS (Android O) would help in figuring out the cause. I would try to test it out on a One Plus with Android O tomorrow. Meanwhile, if you have any other device with Android O available, please test on that.

**Comment by MrVipinVijayan** [comment]
I tried on another device (Mi A1) with oreo, the same issue happens there
as well.

**Comment by nishkarsh** [comment]
@MrVipinVijayan Could you provide logcat logs when this occurs? Also, again, if you could share the sample code.

**Comment by MrVipinVijayan** [comment]
I am using your code only.

**Comment by nishkarsh** [comment]
@MrVipinVijayan This is a library and there is a set of APIs that you could call. It would help in knowing the APIs that you are calling and the order in which they are being called. Logs would help in knowing what Android might be doing at the time you connect to WiFi or even the reason for switching back to the previous network.

**Comment by MrVipinVijayan** [comment]
LOGS

04-25 11:15:30.385 2533-2533/wifitest.com.wifitest D/WifiUtil: Registered
for WiFi State broadcast
04-25 11:15:30.415 2533-2533/wifitest.com.wifitest D/WifiUtil:
Un-registered for WiFi State broadcast
04-25 11:15:30.415 2533-2533/wifitest.com.wifitest D/WifiManager: Wifi is
enabled, starting scan
04-25 11:15:30.417 2533-2533/wifitest.com.wifitest D/WifiUtil: Registered
for WiFi Scan results broadcast
04-25 11:15:35.273 2533-2533/wifitest.com.wifitest I/WifiHelper: Available
SSIDs count: 39
04-25 11:15:35.274 2533-2533/wifitest.com.wifitest D/WifiUtil:
Un-registered for WiFi Scan results broadcast
04-25 11:15:35.275 2533-2533/wifitest.com.wifitest I/WifiManager: SSID
wifi_to_be_connected is available.
04-25 11:15:35.279 2533-2533/wifitest.com.wifitest I/WifiUtil: Currently
active network is not quincyguest, would bind the app to use this when
available
04-25 11:15:35.284 2533-4180/wifitest.com.wifitest I/WifiUtil: Network is
Available. Network Info: [type: WIFI[], state: CONNECTED/CONNECTED, reason:
(unspecified), extra: "my_wifi_old", failover: false, available: true,
roaming: false]
04-25 11:15:35.285 2533-2533/wifitest.com.wifitest D/WifiUtil: Registered
for Network State broadcast
04-25 11:15:35.295 2533-2533/wifitest.com.wifitest D/WifiHelper: Priority
assigned to configuration is 1
04-25 11:15:35.297 2533-2533/wifitest.com.wifitest D/WifiHelper: networkId
assigned while adding network is -1
04-25 11:15:35.308 2533-2533/wifitest.com.wifitest D/WifiHelper: networkId
of existing network is 2
04-25 11:15:35.317 2533-2533/wifitest.com.wifitest D/WifiUtil:
Un-registered for Network State broadcast
04-25 11:15:38.033 2533-4180/wifitest.com.wifitest I/WifiUtil: Network is
Available. Network Info: [type: WIFI[], state: CONNECTED/CONNECTED, reason:
(unspecified), extra: "wifi_to_be_connected", failover: false, available:
true, roaming: false]
04-25 11:15:38.040 2533-4180/wifitest.com.wifitest I/WifiUtil: Bound
application to use wifi_to_be_connected network
04-25 11:15:38.040 2533-4180/wifitest.com.wifitest E/MainActivity:
onConnectionEstablished




Today it took some more time to disconnect and switch to previous network,
about 1 min.

**Comment by nishkarsh** [comment]
@MrVipinVijayan I tried a sample app on One Plus 5 with Oreo and I was not able to reproduce this issue. The setup had 2 open networks and the app displayed the list of networks, tapping on one would initiate the connection. In my case, I was able to connect to the one I selected. I did try doing it multiple times and it worked each time without any issue.

I would upload the sample app on Github very soon for you and for anyone else who wants to get an idea of how to try out this library. Do test it out with sample app once and let me know if the issue exists even then.

**Comment by MrVipinVijayan** [comment]
Yeah. Sure. I will check with your new code.

**Comment by nishkarsh** [comment]
@MrVipinVijayan I have uploaded a quick demo of the library. Please check it out here: https://github.com/nishkarsh/AndroidWifiConnectSample

**Comment by nishkarsh** [comment]
@MrVipinVijayan Any updates on this? Did you try it out with the sample app?

**Comment by MrVipinVijayan** [comment]
No, it didn't work..still the same problem. Drops to previous WiFi after
10-15 seconds.
